gorilla/client.go

```diff
@@ -0,0 +1,301 @@
+// +build go1.7
+
+package gorilla
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptrace"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+type contextKey int
+
+const (
+	keyTracer contextKey = iota
+)
+
+const defaultComponentName = "github.com/gorilla/mux"
+
+// Transport wraps a RoundTripper. If a request is being traced with
+// Tracer, Transport will inject the current span into the headers,
+// and set HTTP related tags on the span.
+type Transport struct {
```

Comment:
so is this completely replicating functionality from go-stdlib? Could we not just wrap that one and only have gorilla-specific piece here? That way if we want to improve the general http handling we only need to fix it in one place, in go-stdlib

Reply:
Good idea, Will update it in next commit.